### PR TITLE
Fix dependency graph prior to Rust 2018 update

### PR DESCRIPTION
**Pull request by secretfader**
This PR fixes dependency resolution with the toolchain packaged alongside newer versions of Rust (and probably has something to do with my changes in the rust-av repo). I'll update code practices and style in a separate PR, but if this passes CI it should be a rather minor upgrade to go along with my other update PRs.

### CODE REVIEW

**Comment by lu-zero** [comment]
av-data released should work with it. (and I should publish av-format as well)

**Comment by secretfader** [comment]
That would be handy to have av-data on crates.io. I can update to the published version in this PR rather than the star version.

**Comment by lu-zero** [comment]
https://crates.io/crates/av-data is published, av-format not yet.

**Comment by secretfader** [comment]
Thanks for clarifying. I must've accidentally transposed the crate names when composing my response. 

Assuming CI passes for this crate (waiting on AppVeyor as of this writing), do you want to publish `av-format` before we conclude this PR?

**Comment by lu-zero** [comment]
I published it now, this PR can be amended :)

**Comment by secretfader** [comment]
I updated the versions for `av-data` and `av-format`, while tidying a few other dependencies that were using carat requirements to standardize dependency format. (It's possible that should've been a separate commit, and can still be if it needs to.)

There are a few compilation warnings to be dealt with once this PR goes in.

File: Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,7 @@ pretty_env_logger = "0.3"
 
 [workspace]
 members = ["tools"]
+
```
Comment (lu-zero) [inline Cargo.toml]: Ideally we could drop this hunk now.

Comment (secretfader) [reply Cargo.toml]: I removed the workspace block just now.

Comment (lu-zero) [reply Cargo.toml]: I meant the patch one.

File: Cargo.toml [hunk 2]
```diff
@@ -11,16 +11,17 @@ name = "matroska"
...
+av-data = "0.1"
+av-format = "0.1"
 circular = "0.2"
 log = "0.4"
 
 [dev-dependencies]
 quickcheck = "0.8"
 pretty_env_logger = "0.3"
 
-[workspace]
```
Comment (lu-zero) [inline Cargo.toml]: this should stay.

File: Cargo.toml [hunk 3]
```diff
@@ -11,16 +11,17 @@ name = "matroska"
...
-members = ["tools"]
+[patch.crates-io]
+av-data = { git = "https://github.com/rust-av/rust-av", branch = "master" }
```
Comment (lu-zero) [inline Cargo.toml]: this should not be needed anymore.

File: Cargo.toml (hunk 2 above)
Comment (secretfader) [reply Cargo.toml]: It seemed like an odd change to make, but what you needed wasn't clear to me from your comment. Thanks for clarifying.

**Comment by secretfader** [comment]
We got to the destination, so it's all good. 

It's not critical, but one thing to consider is how it may be useful to override Crates.io for this repo in development and CI environments, so any submitted PRs run against `master` rather than a released version. This is how other teams I've worked on dealt with growing pains associated with rapid development in an ecosystem of crates. Anecdotally, I've noticed this technique simplifying processes for maintainers and revealing bugs that might be hidden from new contributors.

**Comment by lu-zero** [comment]
I tend to prefer having `.cargo/config` for this purpose since it makes relatively straightforward to cut releases.

**Pull request merged**